lib/zlib.js

```diff
@@ -170,82 +192,84 @@ function Zlib(opts, mode) {
   var strategy = Z_DEFAULT_STRATEGY;
   var dictionary;
 
-  if (typeof mode !== 'number')
-    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'mode', 'number');
-  if (mode < DEFLATE || mode > UNZIP)
-    throw new errors.RangeError('ERR_OUT_OF_RANGE', 'mode');
+  // The Zlib class is not exported to user land, the mode should only be
+  // passed in by us.
+  assert(typeof mode === 'number');
+  assert(mode >= DEFLATE && mode <= UNZIP);
 
   if (opts) {
     chunkSize = opts.chunkSize;
-    if (chunkSize !== undefined && chunkSize === chunkSize) {
-      if (chunkSize < Z_MIN_CHUNK || !Number.isFinite(chunkSize))
-        throw new errors.RangeError('ERR_INVALID_OPT_VALUE',
-                                    'chunkSize',
-                                    chunkSize);
+    if (validateFiniteNumber(chunkSize, 'options.chunkSize')) {
+      if (chunkSize < Z_MIN_CHUNK) {
+        throw new errors.RangeError('ERR_OUT_OF_RANGE', 'options.chunkSize',
+                                    `>= ${Z_MIN_CHUNK}`, chunkSize);
+      }
     } else {
       chunkSize = Z_DEFAULT_CHUNK;
     }
```

Comment:
Non blocking: A very personal point about readability: I feel like it would be better to write it as:

```js
if (!validateFiniteNumber(...) {
  chunkSize = Z_...
} else if (chunkSize < Z_MIN_CHUNK) {
  throw new ...
}
```